### PR TITLE
feat: progress event

### PR DESCRIPTION
**Pull request by csiyang**
# Description

Add progress events to video

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/26787374/todos/4816010650)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Tests pass

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged into main

### CODE REVIEW

File: libs/journeys/ui/src/components/Video/Video.tsx [hunk 1]
```diff
@@ -176,6 +181,7 @@ export function Video({
                 <VideoTrigger player={playerRef.current} {...option} />
               )
           )}
+          {console.log(startAt, endAt)}
```
Comment (csiyang) [inline libs/journeys/ui/src/components/Video/Video.tsx:184]: remove

**Review by tataihono — DISMISSED** [review]
we shouldn't care if events here get sent twice because a user is scrubbing backwards and forwards. We do want to know if there are sections people are rewatching and this would mask that.

I think this can be vastly simplified by removing the refs. Also end should be set to duration if no endAt.

File: libs/journeys/ui/src/components/Video/VideoEvents/VideoEvents.tsx [hunk 2]
```diff
@@ -79,82 +93,183 @@ export function VideoEvents({
...
         }
       })
-    })
+    }
+    player.on('pause', pauseListener)
+    return () => player.off('pause', pauseListener)
+  }, [player, videoPauseEventCreate, blockId])
 
-    player.on('ended', () => {
+  // COMPELTE event
```
Comment (GeronimoJohn) [inline libs/journeys/ui/src/components/Video/VideoEvents/VideoEvents.tsx:167]: Spellingh

**Comment by tataihono** [comment]
closing in favor of https://github.com/JesusFilm/core/pull/629

**Pull request closed without merging**